src/sources/splunk_hec.rs

```diff
@@ -0,0 +1,954 @@
+use lazy_static::lazy_static;
+use serde::{de, Deserialize, Serialize};
+use serde_json::{de::IoRead, json, Deserializer, Value};
+use std::sync::{Arc, Mutex};
+use std::{
+    io::Read,
+    net::{Ipv4Addr, SocketAddr},
+};
+use stream_cancel::{Trigger, Tripwire};
+use string_cache::DefaultAtom as Atom;
+
+// Event fields unique to splunk_hec source
+lazy_static! {
+    pub static ref CHANNEL: Atom = Atom::from("channel");
+    pub static ref INDEX: Atom = Atom::from("index");
+    pub static ref SOURCE: Atom = Atom::from("source");
+    pub static ref SOURCETYPE: Atom = Atom::from("sourcetype");
+}
+
+/// Cashed bodies for common responses
+mod splunk_response {
+    use bytes::Bytes;
+    use lazy_static::lazy_static;
+    use serde_json::{json, Value};
+
+    fn json_to_bytes(value: Value) -> Bytes {
+        serde_json::to_string(&value).unwrap().into()
+    }
+
+    lazy_static! {
+        pub static ref INVALID_AUTHORIZATION: Bytes =
+            json_to_bytes(json!({"text":"Invalid authorization","code":3}));
+        pub static ref MISSING_CREDENTIALS: Bytes =
+            json_to_bytes(json!({"text":"Token is required","code":2}));
+        pub static ref NO_DATA: Bytes = json_to_bytes(json!({"text":"No data","code":5}));
+        pub static ref SUCCESS: Bytes = json_to_bytes(json!({"text":"Success","code":0}));
+        pub static ref SERVER_ERROR: Bytes =
+            json_to_bytes(json!({"text":"Internal server error","code":8}));
+        pub static ref SERVER_SHUTDOWN: Bytes =
+            json_to_bytes(json!({"text":"Server is shuting down","code":9}));
+        pub static ref UNSUPPORTED_MEDIA_TYPE: Bytes =
+            json_to_bytes(json!({"text":"unsupported content encoding"}));
+        pub static ref NO_CHANNEL: Bytes =
+            json_to_bytes(json!({"text":"Data channel is missing","code":10}));
+    }
+}
+
+/// Accepts HTTP requests.
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct SplunkConfig {
+    /// Local address on which to listen
+    #[serde(default = "default_socket_address")]
+    address: SocketAddr,
+    /// Splunk HEC token
+    token: String,
+}
+
+fn default_socket_address() -> SocketAddr {
+    SocketAddr::new(Ipv4Addr::new(0, 0, 0, 0).into(), 8088)
+}
+
+#[typetag::serde(name = "splunk_hec")]
+impl SourceConfig for SplunkConfig {
+    fn build(
+        &self,
+        _: &str,
+        _: &GlobalOptions,
+        out: mpsc::Sender<Event>,
+    ) -> crate::Result<super::Source> {
+        let (trigger, tripwire) = Tripwire::new();
+
+        let source = Arc::new(SplunkSource::new(self, out, trigger));
+
+        let service = move || future::ok::<_, String>(Connection::new(source.clone()));
+
+        // Build server
+        let server = Server::bind(&self.address)
+            .serve(service)
+            .with_graceful_shutdown(tripwire)
+            .map_err(|error| error!(message="Splunk HEC source stopped, because of error", %error));
```

Comment:
```suggestion
            .map_err(|error| error!(message="Splunk HEC source stopped.", %error));
```

the `error!` macro should be enough to inform users it was an error :)

Reply:
True :D